### ISSUE TITLE
[ASCollectionNode / ASCollectionDelegateFlowLayout] Unable to make a header disappear

### ISSUE THREAD

**Issue by flovouin** [TextureGroup/Texture#602]
Hi guys,

I've recently started to use headers in a `ASCollectionNode`, and it works great. However I also need to make some of them disappear later on, and this doesn't seem to work.

From what I read around the interwebz, returning a size of zero in `sizeRangeForHeaderInSection` is enough to make the header disappear. This does not seem to be the case with `ASCollectionNode`.

Here is the sample project: https://github.com/flovouin/Texture-CollectionHeaders
At first the header is not visible. After 2 seconds, it appears. After two more seconds, I try to make it disappear, but nothing changes. I can successfully change its size, but I can't set it to zero / hide it.

What I tried:
- Returning an exact size range with a height of 0 in `sizeRangeForHeaderInSection`.
- Returning an exact size range of 0 in `sizeRangeForHeaderInSection`. (Just to be sure.)
- Return an empty array in `supplementaryElementKindsInSection` once I want to make the header disappear.

By observing when `nodeForSupplementaryElementOfKind` is called, it looks like the correct logic is already implemented, as the node is only requested once the size is non-zero. However setting this size back to zero later on does not make the header disappear.

Am I missing something or is this not implemented?

Cheers,

Flo

**Comment by afrederick1** [comment]
There is a temporary fix listed here: https://github.com/TextureGroup/Texture/issues/123

TL;DR You can't return 0 for the size, you can return something like 0.5

Also when you're ready to change the size of the header, the most efficient way to do it is by using an invalidation context:
`
let context = UICollectionViewFlowLayoutInvalidationContext()
context.invalidateFlowLayoutDelegateMetrics = true
collectionNode.collectionViewLayout.invalidateLayout(with: context)
`

**Comment by flovouin** [comment]
Thanks, @afrederick1 ! I had figured out the non zero height on my own, but this causes side effects in the layout (spacing between the "hidden" header and the items). Nothing I can't work around, just not super convenient. :-)
I didn't know about this way to update the header height, thanks a lot!

**Comment by nguyenhuy** [comment]
@flovouin Right now supplementary nodes are (aggressively) repopulated whenever one or more items are inserted or deleted. You're correct that we don't have an API to re-query and update existing supplementary nodes. It's certainly inconvenient. Please feel free contribute a new API and, even better, implementation proposal. Cheers!
